### PR TITLE
Replace CWalletDB::ReadKeyValue with CWallet::LoadKeyValue

### PR DESCRIPTION
**Pull request by pstratem**
This moves logic out of CWalletDB

### CODE REVIEW

File: src/wallet/walletdb.h [hunk 1]
```diff
@@ -41,6 +41,24 @@ enum DBErrors
     DB_NEED_REWRITE
 };
 
+class CWalletScanState {
```
Comment (laanwj) [inline src/wallet/walletdb.h]: I'd say this belongs in wallet.h, not walletdb.h, as it's not used in walletdb anymore?

Comment (pstratem) [inline src/wallet/walletdb.h]: It's still used in CWalletDB::LoadWallet

Comment (sipa) [inline src/wallet/walletdb.h]: Splitting it up across files is pretty ugly in that case. Is there a next step PR in the pipeline to fix that?

Comment (pstratem) [inline src/wallet/walletdb.h]: there isn't currently but there will be

**Comment by pstratem** [comment]
@sipa latest commit completely removes CWalletDB::LoadWallet

Still cant move CWalletScanState because of CWalletDB::Recover

**Comment by sipa** [comment]
Wouldn't it be possible to create a simple CWalletLoader class in walletdb.h, with only virtual functions, have CWallet implement CWalletLoader and then let CWalletDB::LoadWallet take a CWalletLoader* argument?

**Comment by pstratem** [comment]
@sipa I guess, but why?

CWalletDB really should just be doing serialization

**Comment by sipa** [comment]
@pstratem Serialization is in wallet.h (by virtue of the SerializationOp methods in the data structure). Walletdb's responsibility is knowing the file format (what keys correspond to what, interacting with BDB, ...). It seems that your pull here is partially moving the knowledge of the file format to CWallet itself.

By introducing a CWalletLoader interface, you can just move the knowledge for how to load entries to the wallet itself, while leaving the format definition in walletdb.

**Comment by laanwj** [comment]
I agree with @sipa here. The goal is not to move everything related to the wallet and knowledge of everything into CWallet. This appears to be happening lately: part of initialization was moved into CWallet, argument parsing was moved into CWallet, and so on.

Splitting up a "god class" later is quite difficult, so I'd also prefer not going through that and introducing a CWalletLoader which is concerned with the file format.

**Comment by laanwj** [comment]
Closing this - would need extensive rebasing, and it seems from above discussion that there was no agreement on this approach.
Let me know if it should be re-openened.

**Pull request closed without merging**